File: app/src/main/java/net/sanic/Kayuri/ui/main/player/ModifyGestureDetector.java
package net.sanic.Kayuri.ui.main.player;

import android.content.Context;
import android.view.GestureDetector;
import android.view.MotionEvent;

public class ModifyGestureDetector extends GestureDetector {

    MyGestureListener myGestureListener;

    public ModifyGestureDetector(Context context, OnGestureListener listener) {
        super(context, listener);
        init(listener);
    }

    void init(GestureDetector.OnGestureListener listener){
        if (listener instanceof MyGestureListener){
            myGestureListener = (MyGestureListener) listener;
        }
    }

    //u can write something more complex as long as u need
    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        if(ev.getAction() == MotionEvent.ACTION_UP
                && myGestureListener != null){
            myGestureListener.onUp(ev);
        }
        return super.onTouchEvent(ev);
    }

    public interface MyGestureListener{
        void onUp(MotionEvent ev);
    }}

File: app/src/main/java/net/sanic/Kayuri/utils/exoplayer/ModifyGestureDetector.kt
package net.sanic.Kayuri.utils.exoplayer

import android.content.Context
import android.view.GestureDetector
import android.view.MotionEvent

class ModifyGestureDetector(context: Context?, listener: OnGestureListener) :
    GestureDetector(context,listener) {
    var myGestureListener: MyGestureListener? = null
    fun init(listener: OnGestureListener?) {
        if (listener is MyGestureListener) {
            myGestureListener = listener
        }
    }

    //u can write something more complex as long as u need
    override fun onTouchEvent(ev: MotionEvent): Boolean {
        if (ev.action == MotionEvent.ACTION_UP
            && myGestureListener != null
        ) {
            myGestureListener!!.onUp(ev)
        }
        return super.onTouchEvent(ev)
    }

    interface MyGestureListener {
        fun onUp(ev: MotionEvent?)
    }

    init {
        init(listener)
    }
}